### ISSUE TITLE
Custom Signup Redirect with Autosignup

### ISSUE THREAD

**Issue by invertium** [pennersr/django-allauth#1655]
During the discussion on this [stackoverflow question](http://stackoverflow.com/questions/42040601/add-second-sign-up-screen-to-social-sign-up-allauth/42621658?noredirect=1#42621658) it became clear to me that if auto-signup is enabled, there is no way to override the method that decides where to go after the signup since its in the socialaccount.helpers file. (So is the "complete_social_signup method)

Compare to the "normal" signup (see link above) on how easy it is to override that.

Is there a smooth way to do this or is that currently not supported?

If not, I would propose adding an adapter or providing a callable to a setting that does the same as "complete_signup" for all accounts. 

I ended up forking the repo and adding the setting ACCOUNT_SIGNUP_REDIRECT to add my own redirect method. See commit [here](https://github.com/JulianSchneider/django-allauth/commit/ff87a6727d0873fe7db9fda1c4273ddf69a22952).

**Comment by fjsj** [comment]
Had the same issue. Ended up using another library.

**Comment by pennersr** [comment]
I am not sure I understand your scenario. For both normal and auto signup, `get_login_redirect_url()` is called on the adapter. Also, `adapter.login()` is called. And, before auto signup kicks in, `is_auto_signup_allowed()` is invoked. In any of these you are free to inspect & store state, or raise an`ImmediateHttpResponse(HttpResponseRedirect('/your/redirect'))` to redirect as you like.  So I am not sure what you think is missing ...

**Comment by fjsj** [comment]
@pennersr maybe for @JulianSchneider this would work. For my use case I think it wouldn't.

Here is what I have:
- It's a GitHub-related application. Users can only login via GitHub.
- There is a single button in home, both for login and sign up.
- Clicking on it should direct user to different flows, depending if it's login or sign up.

I couldn't find a way to redirect to another page if it was a sign up. AFAIK there is no way to know it's a sign up instead of a login. Could you check this?

**Comment by JulianSchneider** [comment]
@pennersr the point is to have different redirect urls for signup and login. Login should go where the next variable points to (i.e. where the user came from), signup ALWAYS points towards /my-onboarding-url carrying the next parameter for a later redirect. I could not find a way of doing this without forking for both normal and social accounts.

**Comment by nigfinch** [comment]
Hi @JulianSchneider 
Were you able to solve this ?
I am writing an app with a similar requirement.
Social Signup - redirect to onboarding
Social Login - redirects to /next/ value

**Comment by eriktelepovsky** [comment]
+1

**Comment by pennersr** [comment]
@invertium  In order to always redirect to your onboarding URL, you'll need to keep track of state indicating whether or not the account is fully onboarded, this is not something allauth can do as that goes beyond authentication.  While a simple ACCOUNT_SIGNUP_REDIRECT_URL can be added to redirect you elsewhere directly after signing up, you won't end up at that URL the user does not complete the onboarding and tries to login at a later time.

**Comment by pennersr** [comment]
Closed via 8c8ea6ea -- see the additional note in the documentation though, the remark on onboarding still stands.

**Comment by nmacchitella** [comment]
Hey @pennersr I am facing a similar issue here that I think I am unable to solve because of my lack of knowledge of python and django-allauth ( I am a fairly young coder) rather than django-allauth limitation.

I tried to detail the matter in this [stackoverflow](https://stackoverflow.com/questions/70208543/django-allauth-redirect-users-post-sign-up-sing-in-to-a-specific-page-depending) question but I would really appreciate some help. I have been stuck on this problem for quite some time at this point and I am running out of ideas. 

**The issue is the following:**

- I have an invite system in my django app that allow users to invite other people (using their email) to join their team.
- When an invited user clicks on an invitation, they are redirected to an url that carries informations about the specific team the user was invited to.
- Such url calls a view called accept_invitation that handles the following use-cases:

a) If a user is logged-in already AND they are also the invited user (i check the email of the invite), I add the user to the team and redirect them to the team's page.

b) If a user is logged-in already BUT they are not the invited user, I redirect them to a page where I invite them to log-in with a different account.

c) If a user is logged-out, I redirect them to a page where I invite them to log-in and/or sign-up to continue.

Use-case a) works just fine but use-case b) and c) I had to figure out how to propagate the informations of the invite to the following screens and past the login.

My initial idea was to add the parameters of the invitation to the request.session so that after the login-in I could have retrieved the informations from the request inside a MyAccountAdapter(DefaultAccountAdapter) adapter I defined then send the users where they had to go.

However, I am doing something wrong and MyAccountAdapter is not getting called. I am not sure why. I think it's because of another adapter MySocialAccountAdapter I made that maybe is messing things up (and I am not even sure if it's doing what it's supposed to do honestly).

[code in the stackoverflow link]
